### ISSUE TITLE
After displaying completions, the entire interaction pane is empty

### ISSUE THREAD

**Issue by lokedhs** [McCLIM/McCLIM#651]
To reproduce, open an interaction pane and type a few commands. Then right-click to display the completions and select an option. After this, the entire history has disappeared from the pane.

After some investigation I discovered that the pane is not actually cleared, but rather that all content is scrolled up outside the visible view. The reason is as follows:

https://github.com/mcclim/McCLIM/blob/master/Core/clim-core/input-editing.lisp#L736

In the code above, the function `PRINT-POSSIBILITIES` displays the available alternatives. This output is stored in a special output record called the `TYPEOUT-RECORD` (which is a slot in `STANDARD-INPUT-EDITING-MIXIN`).

All output records below the typeout record is then scrolled down by the height of the typeout record. The effect of this is that the completions is inserted above the prompt.

After the selection is made, the function `CLEAR-TYPEOUT` is called, which removes the typeout record, and scrolls everyting below it up by the height of the typeout record. This causes the completions box to be removed, and everything below it (i.e. the prompt and whatever input the user have typed) to be scrolled up to where it was before the completion box was displayed.

The problem is that during the call to `REPAINT-SHEET` the output record's Y position has been changed to 0, causing the output record height to be the height of the entire interaction pane, which causes all output records to be scrolled up (not just the ones below the completion box). All output records are now scrolled up to negative coordinates so that they are not visible anymore, giving the appearance of the entire interaction pane having been cleared.

This call happens here: https://github.com/mcclim/McCLIM/blob/master/Core/clim-core/input-editing.lisp#L274

It is as of yet unclear where the output record's y coordinate gets changed, but I have discovered that the issue is triggered by the double borders in `PRINT-POSSIBILITIES`. Here's part of its implementation:

```
(with-input-editor-typeout (stream :erase t)
    (surrounding-output-with-border (stream :shape :drop-shadow :background +cornsilk1+)
      (surrounding-output-with-border (stream :shape :rectangle)
        (let ((ptype `(completion ,possibilities)))
          ...
```

If either of the two borders are removed, the problem goes away. The issue must therefore have something to do with the interaction between these two calls.

**Comment by lokedhs** [comment]
Some more details:

The call to `PRINT-POSSIBILITIES` results in a call to `INVOKE-WITH-EDITOR-TYPEOUT`. When I log the Y coordinate of the output record before and after the call to `REPAINT-SHEET`, I can see this value changing some time during that call (and because the height is equal to y2-y1, this is why the height changes).

This is the location of the call that causes the output record to change: https://github.com/mcclim/McCLIM/blob/master/Core/clim-core/input-editing.lisp#L274

**Comment by dkochmanski** [comment]
The problem originates in `replay-output-record` specialized on the bordered-output. In order to update the border, history is changed by recreating under/over records. This change was introduced in order to fix #165. Related to this one is issue #554.

I have three ideas for a fix:

1. implement `note-output-record-child-changed` to work on all output records (it has some implementation only for displayed output records and I think it is incorrect). Then specialize bordered-output-record for it. This seems to be the correct way of doing this, but requires a considerable abount of work and careful code/spec analysis.
2. Specialize output-record-position for bordered-output-record to compute position dynamically. That should allow us to get rid of output history modification and I think it is correct. It should take less time and testing than 1.
3. make replay-output-record update positions in a correct order. This is more a hack than a solution. We should condamn changing output record history from replay-output-record as invalid behavior to avoid tripping people that repaint-sheet affects their records (and for performance reasons).
